types/event.go

```diff
@@ -64,11 +65,29 @@ func (t *Trace) AddSpan(sp *Span) {
 // GetSpans returns the list of spans in this trace
 func (t *Trace) GetSpans() []*Span {
 	return t.spans
+}
+
+func (t *Trace) GetSamplerKey() (string, bool) {
+	if IsLegacyAPIKey(t.APIKey) {
+		return t.Dataset, true
+	}
 
+	env := ""
+	for _, sp := range t.GetSpans() {
+		if sp.Event.Environment != "" {
+			env = sp.Event.Environment
```

Comment:
assuming all events have the same environment, we can just break out of the loop after we find the first one?